AIMSICD/src/main/java/com/secupwn/aimsicd/AndroidIMSICatcherDetector.java

```diff
@@ -56,6 +59,30 @@ public void onCreate() {
         addModule(new AndroidLoggerModule());
         addModule(OkHttpModule.withCache(this));
         super.onCreate();
+
+        RealmConfiguration realmConfiguration = new RealmConfiguration.Builder(this)
+                .deleteRealmIfMigrationNeeded()
+                .build();
+
+        Realm.setDefaultConfiguration(realmConfiguration);
+        realm = Realm.getDefaultInstance();
```

Comment:
Not closing this realm instance is theoretically a resource leak. It isn't dangerous as such and will get closed properly when the app is killed, but having it around will increase the likelihood of the app getting killed while it is in the background. 

You probably want to close it in onSuccess/OnError of the transaction below.